src/components/structures/MessagePanel.js

```diff
@@ -368,6 +369,15 @@ module.exports = React.createClass({
         this.eventNodes[eventId] = node;
     },
 
+    // once dynamic content in the events load, make the scrollPanel check the
+    // scroll offsets.
+    _onWidgetLoad: function() {
+        var scrollPanel = this.refs.scrollPanel;
+        if (scrollPanel) {
+            scrollPanel.checkScroll();
```

Comment:
because `checkScroll` is called from `componentDidUpdate`, so you'd end up with a lot of re-renders happening each time anything in the panel updated.

just `forceUpdate` the `ScrollPanel` instead.